### DISCUSSION TITLE
Why is hibernation available in LMDE 6 and not in Mint 21.3?

### DISCUSSION THREAD

**Discussion by marco4mp** [linuxmint/discussions#479] [Questions] [answered]
Why is hibernation available in LMDE 6 and not in Mint 21.3?

Anyway the Hibernation available in LMDE 6 causes a system freeze.

**Comment by claudiux** [comment, marked as answer]
> Why is hibernation available in LMDE 6 and not in Mint 21.3?

Because Mint becomes from Ubuntu and Ubuntu doesn't activate hibernation by default.

> 
> Anyway the Hibernation available in LMDE 6 causes a system freeze.

You must indicate which swap partition you want to use to resume the system. This is how to do:

1. Determine the UUID of tour swap partition:
`swapon --show`
returns a response like:
```
NAME      TYPE       SIZE USED PRIO
/dev/sda4 partition 30,5G   0B   -2
```

`cat /etc/fstab | grep swap`
returns something like:
`UUID=2f59a476-3912-434c-8362-fa7c1b797a70 none            swap    sw              0       0`
Copy the "UUID=2f59a476-3912-434c-8362-fa7c1b797a70" part of this line.

2. Activate your swap partition for hibernation:
`sudo xed /etc/default/grub`
Change the line beginning with "GRUB_CMDLINE_LINUX_DEFAULT" from:
`GRUB_CMDLINE_LINUX_DEFAULT="quiet splash"` to:
`GRUB_CMDLINE_LINUX_DEFAULT="quiet splash atkbd.reset resume=UUID=2f59a476-3912-434c-8362-fa7c1b797a70"` (pasting the "UUID=..." part).

Save "/etc/default/grub" and close xed. Then run the command:
`sudo update-grub`
When finished, reboot your computer.

3. Hibernate!
`systemctl hibernate`
(enter your password)

 The display may disappear and reappear before the computer is switched off.

To bring your computer out of hibernation, use its power button. You'll be prompted to resume from the partition whose UUID is... the one you entered.

**Reply by marco4mp** [reply]
Oh okay, that makes sense.

Thanks for the help! But I don't need to use hibernate. Do you know how I can disable it so that it doesn't appear as an option every time I want to turn off the system?

Wouldn't it be good to maintain consistency between both versions? (Disable it by default also in the Debian version).

**Comment by claudiux** [comment]
If you want a configurable applet to stop or restart your computer (and so on), you can use `Exit@claudiux`: https://cinnamon-spices.linuxmint.com/applets/view/386

**Comment by 100savage** [comment]
This is how I disable hibernation
First I create a file called disable-hibernate.pkla with the following information in it
```
[Disable hibernate (upower)]
Identity=unix-user:*
Action=org.freedesktop.upower.hibernate
ResultActive=no
ResultInactive=no
ResultAny=no

[Disable hibernate (logind)]
Identity=unix-user:*
Action=org.freedesktop.login1.hibernate
ResultActive=no

[Disable hibernate for all sessions (logind)]
Identity=unix-user:*
Action=org.freedesktop.login1.hibernate-multiple-sessions
ResultActive=no
```
Then I copy that file to the proper location using sudo. Make sure you use the correct location to the disable-hibernate.pkla file.
```
sudo cp location/of/the/disable-hibernate.pkla /etc/polkit-1/localauthority/90-mandatory.d/
```

Reboot your system. Your system will no longer go into hibernation.
as an extra bonus. I also prevent my system from being able to be suspended with the disable-suspend.pkla file.
```
[Disable suspend (upower)]
Identity=unix-user:*
Action=org.freedesktop.upower.suspend
ResultActive=no
ResultInactive=no
ResultAny=no

[Disable suspend (logind)]
Identity=unix-user:*
Action=org.freedesktop.login1.suspend
ResultActive=no

[Disable suspend for all sessions (logind)]
Identity=unix-user:*
Action=org.freedesktop.login1.suspend-multiple-sessions
ResultActive=no
```
It's the same process as the disable hibernation.

**Comment by marco4mp** [comment]
I think hibernation should not be enabled by default in LMDE as it brings more problems than benefits.

**Comment by MatzFan** [comment]
Hibernate on LMDE 6 was very useful to me as I run it on an XPS laptop and Dell in their infinite wisdom have removed the deep (s3) sleep option in their laptop firmware - `s2idle` is the only sleep option available. This caused some [consternation](https://www.dell.com/community/en/conversations/linux-general/xps-13-9310-ubuntu-deep-sleep-missing/647f8daff4ccf8a8dee4f308) with users. Apparently this is at Microsoft's request so their Modern Standby is used. My battery used to drain on suspend, but now I can leave it unattended for days and resume from hibernate. Works with full disk encryption too. Your applet is great @claudiux.
